projects/components/src/table/columns/table-edit-columns-modal.component.ts

```diff
@@ -77,4 +78,23 @@ export class TableEditColumnsModalComponent {
   public onCancel(): void {
     this.modalRef.close();
   }
+
+  public onResetToDefault(): void {
+    this.editColumns = this.filterMetadaDataColumnsAndOrderVisible(this.modalData.defaultColumns);
```

Comment:
Is this enough? How are we propagating this change to the parent table component?

Reply:
got it. But still, we would need to do a deep clone at L58. Otherwise, any selection/deselection will get applied to the same objects.

Reply:
Not clear why I need to do the clone-deep, anyways I'm passing corresponding columns from the modal creator, if I click on cancel I'm not having any issues (already tested) and if I need to use the clone-deep I should apply it in line L58 and also L83. Probably better to add it in filterMetadaDataColumnsAndOrderVisible function. Using clone deep is generating a loss in performance, I would like to avoid it

Reply:
discussed on slack.

Reply:
updated (to prevent future issues)